src/js/components/InfiniteScroll/InfiniteScroll.js

```diff
@@ -177,7 +177,7 @@ const InfiniteScroll = ({
       }
     }, 100);
     return () => clearTimeout(timer);
-  }, [renderPageBounds, show, step]);
+  }, [renderPageBounds, scrollShow, show, step]);
```

Comment:
Unless you are thinking this fix is important and related to this PR functionality, here is a quote from https://github.com/grommet/grommet/pull/5289#discussion_r641842233

>I'm concerned that will have unintended side effects, as this effect calls setScrollShow, which will cause scrollShow to change.

Also linking to https://github.com/grommet/grommet/issues/5317

Reply:
Hi @ShimiSun, this change is only related in that the pre-commit hook prevented me from pushing unless I either fixed the linting error or used the `--no-verify` flag while pushing. As your pre-commit hook specifically says, "This is ill-advised since the commit is broken." I'm happy to revert that change, if you prefer to address this by other means.

I've just reverted that change & pushed. It should be ready to review now, please.